lucene/core/src/java/org/apache/lucene/util/graph/GraphTokenStreamFiniteStrings.java

```diff
@@ -124,6 +126,15 @@ public boolean hasSidePath(int state) {
         .toArray(Term[]::new);
   }
 
+  /**
+   * Returns the list of terms that start at the provided state
+   */
+  public QueryBuilder.TermAndBoost[] getTermsAndBoosts(String field, int state) {
```

Comment:
Given that this class, GraphTokenStreamFiniteStrings deals with List<AttributeSource> (something I did not know when I made a previous comment), and also that TermAndBoost is an inner class to QueryBuilder, I think it's better to put this back into QueryBuilder.  I still think `List<AttributeSource>` is weird and heavyweight but you didn't add it.

Reply:
Yes, let's go back to `AttributeSource` - sorry for the back and forth on this @alessandrobenedetti

Reply:
no worries at all, done in the upcoming commit!